packages/interface/src/components/ImageUpload.tsx

```diff
@@ -1,82 +1,94 @@
 import { useMutation } from "@tanstack/react-query";
 import clsx from "clsx";
 import { ImageIcon } from "lucide-react";
-import { type ComponentProps, useRef, useCallback } from "react";
+import { type ComponentProps, ElementType, forwardRef, useRef, useCallback } from "react";
 import { Controller, useFormContext } from "react-hook-form";
 import { toast } from "sonner";
 
 import { IconButton } from "~/components/ui/Button";
 
+import { PolymorphicRef } from "./ui";
+
 export interface IImageUploadProps extends ComponentProps<"img"> {
   name?: string;
   maxSize?: number;
 }
 
-export const ImageUpload = ({
-  name = "",
-  maxSize = 1024 * 1024, // 1 MB
-  className,
-}: IImageUploadProps): JSX.Element => {
-  const ref = useRef<HTMLInputElement>(null);
-  const { control } = useFormContext();
-
-  const select = useMutation({
-    mutationFn: async (file: File) => {
-      if (file.size >= maxSize) {
-        toast.error("Image too large", {
-          description: `The image to selected is: ${(file.size / 1024).toFixed(2)} / ${(maxSize / 1024).toFixed(2)} kb`,
-        });
-        throw new Error("IMAGE_TOO_LARGE");
-      }
+export const ImageUpload = forwardRef(
+  (
+    {
+      name = "",
+      maxSize = 1024 * 1024, // 1 MB
+      className,
+    }: IImageUploadProps,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    ref: PolymorphicRef<ElementType>,
```

Comment:
If you don't use this ref, you don't need to add it as second param

Reply:
I tried but I am getting a console log error saying that `forwardRef` needs a function that receives two parameters and it is not possible to remove the second one (ref). What I did is change its name to `_` and removed the types